### ISSUE TITLE
AuthorSearch ends up in Error 400 when results are over 5000

### ISSUE THREAD

**Issue by JLoDoesIt** [pybliometrics-dev/pybliometrics#197]
I had an issue while doing an author search
```
from pybliometrics.scopus import AuthorSearch
authors = AuthorSearch("AF-ID(60110693)")
```

I obtain
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jlo/workspace/pub-okr/venv/lib/python3.7/site-packages/pybliometrics/scopus/author_search.py", line 123, in __init__
    download=download, verbose=verbose, **kwds)
  File "/home/jlo/workspace/pub-okr/venv/lib/python3.7/site-packages/pybliometrics/scopus/superclasses/search.py", line 62, in __init__
    api=api, verbose=verbose)
  File "/home/jlo/workspace/pub-okr/venv/lib/python3.7/site-packages/pybliometrics/scopus/superclasses/base.py", line 80, in __init__
    verbose, *args, **kwds)
  File "/home/jlo/workspace/pub-okr/venv/lib/python3.7/site-packages/pybliometrics/scopus/superclasses/base.py", line 163, in _parse
    resp = get_content(url, api, params, *args, **kwds)
  File "/home/jlo/workspace/pub-okr/venv/lib/python3.7/site-packages/pybliometrics/scopus/utils/get_content.py", line 97, in get_content
    raise errors[resp.status_code](reason)
pybliometrics.scopus.exception.Scopus400Error: Exceeds the number of search results
```

Digging in the issue:
The query look like:
https://api.elsevier.com/content/search/author?query=AF-ID(60110693)&apiKey=<YOURKEY>
Result look like:
```
<search-results>
<opensearch:totalResults>5315</opensearch:totalResults>
[...]
<link ref="last" href="https://api.elsevier.com/content/search/author?start=4975&count=25&query=AF-ID%2860110693%29&apiKey=1c3e9e143e5d196ad8a4f1b1f91d3f47" type="application/xml"/>
```

It seems that the maximum number of results an Author search can return is 5000 whatever the number of results available in theory.
Right now the code seems to care only about the totalResult field
https://github.com/pybliometrics-dev/pybliometrics/blob/c05526b3b979abc6bb0a23938c7c1125f0713e3b/pybliometrics/scopus/superclasses/base.py#L64

What do you think?

**Comment by Michael-E-Rose** [comment]
On https://pybliometrics.readthedocs.io/en/stable/classes/AuthorSearch.html we explain the Scopus400Error that you're getting:

> If the number of search results exceeds 5000, which is the API’s maximum number of results returned. The error prevents the download attempt and avoids making use of your API key

What you can do is to set `download=False` and then use `.get_results_size()` on the object. All of that explained at https://pybliometrics.readthedocs.io/en/stable/classes/AuthorSearch.html

**Comment by JLoDoesIt** [comment]
neat lib, reaction time and problem solving, thank you !
I have hacked this line to put a min function so I'm not blocked.
Still it does make the API calls as far as I see and don't store to cache because of the unhandled exception.
I see that your code handle it in certain situations (cursor in parameters) but in mine, it fails this condition and fails in get_content because it tries to collect a pagination beyond 5000.
https://github.com/pybliometrics-dev/pybliometrics/blob/c05526b3b979abc6bb0a23938c7c1125f0713e3b/pybliometrics/scopus/superclasses/base.py#L69

Let me know if I can help further

**Comment by Michael-E-Rose** [comment]
Hmm, that shouldn't be the case. pybliometrics should throw the error if there's more than 5k results.

Are you on version 3.0.0 already?

**Comment by JLoDoesIt** [comment]
Yep doing pip freeze shows 3.0.0
If you have >25 spare API calls, you should be able to reproduce
```
from pybliometrics.scopus import AuthorSearch
authors = AuthorSearch("AF-ID(60110693)")
```

**Comment by Michael-E-Rose** [comment]
Indeed, this is a mistake. I'll fix this ASAP.

**Comment by Michael-E-Rose** [comment]
Fixed with 577001756f4f79428fa9aa04a13b1b75e5b18a9f.
